lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -105,6 +106,44 @@ public IJobSubmissionResult SubmitAndGetJobStatus(JobRequest jobRequest)
             return application.FinalStatus;
         }
 
+        /// <summary>
+        /// Returns all the application reports running in the cluster.
+        /// GetApplicationReports call is very expensive as it is trying 
+        /// fetch information about all the applications in the cluster.
+        /// 
+        /// If this method is called right after submitting a new app then
+        /// that new app might not immidietly result in this list until 
+        /// some number of retries. 
+        /// </summary>
+        /// <returns></returns>
+        public async Task<List<IApplicationReport>> GetApplicationReports()
+        {
+            var appReports = new List<IApplicationReport>();
+            try
+            {
+                var applications = await _yarnClient.GetApplicationsAsync();
+
+                foreach (var application in applications.App)
+                {
+                    appReports.Add(new ApplicationReport(application.Id, application.Name, application.TrackingUrl, 
+                                                         application.StartedTime, application.FinishedTime, 
+                                                         application.RunningContainers, application.FinalStatus));
+
+                    var msg = string.Format("application status {0}, Progress: {1}, trackingUri: {2}, " +
+                                            "Name: {3}, ApplicationId: {4}, State {5}.",
+                                             application.FinalStatus, application.Progress, application.TrackingUI, 
+                                             application.Name, application.Id, application.State);
+                    Logger.Log(Level.Verbose, msg);
```

Comment:
Better yet - implement `Application.ToString()`. that will simplify the logging call a lot:
```C#
Logger.Log(Level.Verbose, "Application report: {0}", application);
```